packages/auth/src/platform_browser/persistence/cookie_storage.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Persistence } from '../../model/public_types';
+
+import {
+  PersistenceInternal,
+  PersistenceType,
+  PersistenceValue,
+  StorageEventListener
+} from '../../core/persistence';
+
+export class CookiePersistence implements PersistenceInternal {
+  static type: 'COOKIE' = 'COOKIE';
+  readonly type = PersistenceType.COOKIE;
+  listeners: Map<StorageEventListener, (e: any) => void> = new Map();
+
+  async _isAvailable(): Promise<boolean> {
+    return navigator.hasOwnProperty('cookieEnabled') ?
```

Comment:
Probably need a check on `typeof navigator !== 'undefined'` just in case. IndexedDB persistence has a check on `if (!indexedDB)` inside a try/catch which I think also works.

Reply:
good call, will do